docker/client.py

```diff
@@ -239,7 +241,7 @@ def _multiplexed_response_stream_helper(self, response):
         # Disable timeout on the underlying socket to prevent
         # Read timed out(s) for long running processes
         socket = self._get_raw_response_socket(response)
-        if six.PY3:
+        if six.PY3 and hasattr(socket, "_sock"):
```

Comment:
Nup, in python3 the socket is SocketIO object and has no settimout.

I'll instead go for

        if hasattr(socket, "settimeout"):
            socket.settimeout(None)
        if hasattr(socket, "_sock") and hasattr(socket._sock, "settimeout"):
            socket._sock.settimeout(None)

I think it's reasonable to expect one of them to always have settimeout and that setting it on both shouldn't have bad effects.

This way we also aren't tying this piece of code to whatever conditions are making this distinction necessary.